src/main/java/interface_adapter/login/LoginPresenter.java

```diff
@@ -0,0 +1,58 @@
+package interface_adapter.login;
+
+import data_access.StockDataAccessInterface;
+import entity.Stock;
+import entity.User;
+import use_case.login.LoginOutputBoundary;
+import use_case.login.LoginOutputData;
+import utility.ServiceManager;
+import utility.ViewManager;
+import view.view_events.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LoginPresenter implements LoginOutputBoundary {
+
+    @Override
+    public void prepareSuccessView(LoginOutputData outputData) {
+        User user = outputData.user();
+        // update username data
+        ViewManager.Instance().broadcastEvent(
+                new UpdateUsernameEvent(user.getUsername())
+        );
+        // update user asset data
+        ViewManager.Instance().broadcastEvent(
+                new UpdateAssetEvent(
+                        user.getPortfolio(),
+                        user.getBalance()
+                )
+        );
+        // update history data
+        ViewManager.Instance().broadcastEvent(
+                new UpdateTransactionHistoryEvent(user.getTransactionHistory())
+        );
+        // update stock data
+        try {
```

Comment:
Firstly, I'm not exactly sure whether we need to send `UpdateStockEvent` here at all, as stock data is automatically update d by `StockMarket`. Even if we want to update stock information during login, we should invoke `StockMarket` to receive stock data instead of invoking data access object directly.

Reply:
Currently I'm not clear how stock market works... If it automatically sends out update stock events then it's fine.